app/src/main/java/com/gruppo4/ringUp/permissions/PermissionsHandler.java

```diff
@@ -20,33 +21,39 @@
     /**
      * Checks if all permissions are granted.
      *
-     * @param permissions The permission(s) to check. It can't be null.
-     * @return true if the app has all permissions granted, false otherwise.
+     * @param context     The target Context. It can't be null.
+     * @param permissions The permission(s) to check.
+     * @return false if the app has at least 1 permission denied, true otherwise.
      * @author Francesco Bau', helped by Alberto Ursino.
      */
-    public static boolean checkPermissions(Context context, @NonNull String[] permissions) {
-        for (String permission : permissions) {
-            if (isGranted(context, permission))
-                return false;
+    public static boolean checkPermissions(@NonNull Context context, @Nullable String[] permissions) {
+        if (permissions != null) {
+            for (String permission : permissions) {
```

Comment:
Inside this method you use the "for-each" type of for statement.

Reply:
Aw sheit...
Thank you for spotting that.

Reply:
Yes, because it checks if all permissions are granted.
And, if String[] permissions is null, it means there's nothing to check.

Reply:
Oh, now I see what do you mean...
Now I've always used a foreach statement.